compiler/passes/src/monomorphization/ast.rs

```diff
@@ -31,7 +31,26 @@ use leo_ast::{
 };
 
 use indexmap::IndexMap;
-use itertools::Itertools;
+use itertools::{Either, Itertools};
+
+impl<'a> MonomorphizationVisitor<'a> {
+    /// Evaluates the given constant arguments if possible.
+    ///
+    /// Returns `Some` with all evaluated expressions if all are constants, or `None` if any argument is not constant.
+    fn try_evaluate_const_args(&mut self, const_args: &[Expression]) -> Option<Vec<Expression>> {
+        let mut const_evaluator = ConstPropagationVisitor::new(self.state, self.program, vec![]);
```

Comment:
Why do we now need to go with an approach where we run const prop in monomorphization? Just for my understanding.